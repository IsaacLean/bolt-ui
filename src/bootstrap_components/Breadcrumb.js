/*
 * BOOTSTRAP BREADCRUMB COMPONENT
 * https://v4-alpha.getbootstrap.com/components/breadcrumb
 */

import React from 'react';

export default class Breadcrumb extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    let breadcrumbItems = [];

    if(this.props.breadcrumbItems && this.props.breadcrumbItems.constructor === Array) {
      breadcrumbItems = this.props.breadcrumbItems.map((listGroupItem, i) => <li className="breadcrumb-item" key={i}>
        {listGroupItem}
      </li>);
    }

    return <ol className="breadcrumb">
      {breadcrumbItems}
    </ol>;
  }
}
