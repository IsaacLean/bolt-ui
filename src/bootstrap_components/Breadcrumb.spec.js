import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Breadcrumb from './Breadcrumb';

test('Empty Breadcrumb snapshot', () => {
  const component = shallow(<Breadcrumb />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('ListGroup with listGroupItems.length == 1 snapshot', () => {
  const component = shallow(<Breadcrumb breadcrumbItems={['Item 1']} />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});


test('ListGroup with listGroupItems.length > 1 snapshot', () => {
  const component = shallow(<Breadcrumb breadcrumbItems={['Item 1', 'Item 2']} />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
