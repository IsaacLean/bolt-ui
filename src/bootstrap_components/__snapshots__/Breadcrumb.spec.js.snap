// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Breadcrumb snapshot 1`] = `
<ol
  className="breadcrumb"
/>
`;

exports[`ListGroup with listGroupItems.length == 1 snapshot 1`] = `
<ol
  className="breadcrumb"
>
  <li
    className="breadcrumb-item"
  >
    Item 1
  </li>
</ol>
`;

exports[`ListGroup with listGroupItems.length > 1 snapshot 1`] = `
<ol
  className="breadcrumb"
>
  <li
    className="breadcrumb-item"
  >
    Item 1
  </li>
  <li
    className="breadcrumb-item"
  >
    Item 2
  </li>
</ol>
`;
